### PR TITLE
Add ctx supported eval command.

### PR DESCRIPTION
**Pull request by Neeef**

### CODE REVIEW

File: modules/basics/eval.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			return result.map(item => ctx.reply(item));
+		} else {
+			return ctx.send(result);
+		}
+	}
+
+	makeResultMessages(result, hrDiff, input = null) {
+		const inspected = util.inspect(result, { depth: 0 })
+			.replace(nlPattern, "\n")
+			.replace(this.sensitivePattern, "--snip--");
```
Comment (rjt-rockx) [inline modules/basics/eval.js]:
```js
.replace(this.sensitivePattern, "--snip--");
```
should be
```js
.replace(this.sensitivePattern(ctx), "--snip--");
```

File: modules/basics/eval.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		this.hrStart = process.hrtime();
+		const result = this.makeResultMessages(this.lastResult, hrDiff, ctx.args.script);
+		if(Array.isArray(result)) {
+			return result.map(item => ctx.reply(item));
+		} else {
+			return ctx.send(result);
+		}
+	}
+
+	makeResultMessages(result, hrDiff, input = null) {
```
Comment (rjt-rockx) [inline modules/basics/eval.js]:
```js
makeResultMessages(result, hrDiff, input = null) {
```
should be
```js
makeResultMessages(result, hrDiff, input = null, ctx) {
```

File: modules/basics/eval.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		try {
+			const hrStart = process.hrtime();
+			this.lastResult = eval(ctx.args.script);
+			hrDiff = process.hrtime(hrStart);
+		} catch(err) {
+			return ctx.send(`Error while evaluating: \`${err}\``);
+		}
+
+		this.hrStart = process.hrtime();
+		const result = this.makeResultMessages(this.lastResult, hrDiff, ctx.args.script);
```
Comment (rjt-rockx) [inline modules/basics/eval.js]:
```js
const result = this.makeResultMessages(this.lastResult, hrDiff, ctx.args.script);
```
should be
```js
const result = this.makeResultMessages(this.lastResult, hrDiff, ctx.args.script, ctx);
```

File: modules/basics/eval.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        const message = ctx.message,
+            client = ctx.client,
+            objects = ctx.client.registry.evalObjects,
+            lastResult = this.lastResult;
+		const doReply = val => {
+			if(val instanceof Error) {
+				ctx.reply(`Callback error: \`${val}\``);
+			} else {
+				const result = this.makeResultMessages(val, process.hrtime(this.hrStart));
```
Comment (rjt-rockx) [inline modules/basics/eval.js]:
```js
const result = this.makeResultMessages(val, process.hrtime(this.hrStart));
```
should be
```js
const result = this.makeResultMessages(val, process.hrtime(this.hrStart), null, ctx);
```

File: modules/basics/eval.js (hunk 2 above)
Comment (Neeef) [reply modules/basics/eval.js]: done

File: modules/basics/eval.js (hunk 4 above)
Comment (Neeef) [reply modules/basics/eval.js]: done :D

File: modules/basics/eval.js (hunk 3 above)
Comment (Neeef) [reply modules/basics/eval.js]: also done

File: modules/basics/eval.js (hunk 4 above)
Comment (Neeef) [reply modules/basics/eval.js]: woop, should be now.

File: modules/basics/eval.js [hunk 5]
```diff
@@ -36,7 +36,7 @@ module.exports = class EvalCommand extends global.utils.baseCommand {
 			if(val instanceof Error) {
 				ctx.send(`Callback error: \`${val}\``);
 			} else {
-				const result = this.makeResultMessages(val, process.hrtime(this.hrStart));
+				const result = this.makeResultMessages(this.lastResult, hrDiff, ctx.args.script, ctx);
```
Comment (rjt-rockx) [inline modules/basics/eval.js]: Double facepalm
You weren't supposed to copy-paste the other line
Did you even look at the comment pls

Comment (Neeef) [reply modules/basics/eval.js]: yep, I noticed something was off and just spent the time prior to trying to find out what.

**Comment by rjt-rockx** [comment]
Merging.

**Pull request merged**